### ISSUE TITLE
no git history found for any files

### ISSUE THREAD

**Issue by ryedin** [jakesankey/git-history#1]
Atom version 0.132.0
git-history version 1.1.2

I just installed this promising looking package but it doesn't really seem to work. For every file I try it on (all with a long git history, trust me), it simply says "No history found for _filename_"

Does it maintain its own internal history and can only show changes since the time I install it? I would assume that not to be the case (as it would drastically reduce the utility of such a package). This just appears to be a bug, possibly an incompatibility with the latest version of Atom? Or, am I doing something wrong?

Thank you, the package looks awesome. Hoping I can get it working here as I've been wanting a quick way to side-by-side file histories for a while now.

**Comment by jakesankey** [comment]
You're correct. It shows the actual git history of the file. Not local history. It definitely sounds like a defect caused by changes in the latest atom version. I will take a look this evening and see if this is the case and if so I will update it ASAP. Thanks for the heads up. 

Jake

**Comment by jakesankey** [comment]
I investigated this issue and was unable to reproduce the problem that you are seeing. On Atom 0.132.0 and v1.1.2 of the plugin, I am not seeing any issue. Are you sure you have git globally installed on your computer? When you type 'git' at the command line does it work?

**Comment by ryedin** [comment]
Yeah I use git all the time. Mac OSX, btw.

I'll try uninstalling and re-installing the package later when I get a chance and let you know if it clears up.

**Comment by jakesankey** [comment]
Any luck on this? I have checked compatibility on atom 133 and 134 as well and all seems OK. Let me know if you are still experiencing issues.

**Comment by aminul-technext** [comment]
Hi, i am having the same problem. My Atom version 0.136.0 and git-history version 1.1.2.
I am running Atom on Mac OSX. 
Atom git-diff is showing the differences but somehow git-history could not.
Thank you, you package looks great..

**Comment by jakesankey** [comment]
I really wish I could help. I am still unable to reproduce your issue. If there is any other information you can provide that may help, please let me know.

**Comment by Joe5** [comment]
@jakesankey I have the same problem using Atom 182-190 + your package 2.1.0 - latest. I know a little js/coffee maybe I could help you to debug it somehow.

**Comment by LeiShi92** [comment]
I have the same problem using Atom 1.07 + Git History 3.01

**Comment by Zylann** [comment]
Same on Atom 1.0.13 Windows 7, none of the commands work (history, blame, diff...), even with the "git" command available by modifying the PATH environment variable. Git history 3.1.0.

**Comment by jakesankey** [comment]
@Joe5 - Yes, please feel free to do some debugging and submit a pull request if you solve anything.

@Zylann - This package is only related to the 'history' command. Blame and diff are other packages.

I'm in a bit of bind in that I don't have access to a windows machine to test this out. I have had other comments on their possibly being an issue with the file path formatting differences between mac and windows and I have a potential fix for this issue. I will try  to get it implemented asap and you all can let me know if you still have issues.

**Comment by jakesankey** [comment]
Sorry, scratch that. My potential fix wont necessarily help this issue afterall. Any debugging help from a windows user would be helpful.

**Comment by ChowSinWon** [comment]
some days ago i had the same issue but i thought it had to do with a major git update on my windows system (win10pro 64bit).
the thing i had to do, to get it running again, was adding the git/bin-folder (in my case C:\Program Files\Git\bin) to the user's environment path-variable.

**Comment by Zylann** [comment]
Good idea! I upgraded my Git (which was 1.8.4) to version 2.5.3.windows.1 (released 3 days ago) and now the history package works. I'll see if it works too on my office's PC.

**Comment by DanielAndreasen** [comment]
Same problem here. I use linux.
- git version: 1.8.5.1
- atom version: 1.1.0 (also didn't work at version 1.0.0)
- git-history version: 3.2.0

Can you provide your versions to see if something is going on there.

**Comment by adamreisnz** [comment]
I am also getting the "Could not retrieve history" issue. Commits for files are showing, but none of the histories can be loaded. I've tried several different files.

Might be worth adding that the affected repository is not a Github repository, but hosted on Bluemix. My Github repo's seem to work fine.

**Comment by michaelw85** [comment]
I've got the same issue and I know why it's not showing up. 

So the thing is our repository is quite large and i'm just working on a part of it. So my atom is set to a folder within the repo, if you use "git history show" it will show the changes (commits) in the dropdown but it's not capable of retrieving the actual changes from the file like other people mentioned. When i open up the repo root in atom and try the same on a file it works like a charm!

Hope this helps @jakesankey

**Comment by disklosr** [comment]
I'm having the same issue. Anyone found a solution for this?

EDIT: Ok I was able to get the commit history of a file to show up. But when I select a revision entry it says "Could not retrieve history for [file name]"

**Comment by meetjey** [comment]
Hi, 

same problem here, with Atom 1.8.0 and git history 3.2.0.
I am at the root directory, and message says "No history found for ...."

But i found something, if i start atom from default cmd.exe on windows 10, it works, if i start it from babun (cygwin shell), it fails. Atom seems to load PATH from the shell context, and it fails with integrated GIT in cygwin.

Could be great to allow to set git path in settings.

**Comment by siddhartha-gadgil** [comment]
The comment of michaelw85 solved my problem. Please put the comment in README or somewhere prominent
